src/routes/registerPublicKeys.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  AuthorizedRequest,
+  Env,
+  ProfileWithId,
+  RegisterPublicKeyRequest,
+  RegisterPublicKeyResponse,
+} from '../types'
+import {
+  KnownError,
+  addProfilePublicKey,
+  getProfileFromPublicKey,
+  incrementProfileNonce,
+} from '../utils'
+import { verifyRequestBody } from '../utils/auth'
+
+export const registerPublicKeys = async (
+  request: AuthorizedRequest<RegisterPublicKeyRequest>,
+  env: Env
+) => {
+  const respond = (status: number, response: RegisterPublicKeyResponse) =>
+    new Response(JSON.stringify(response), {
+      status,
+    })
+
+  const {
+    auth: { publicKey },
+    publicKeys,
+  } = request.parsedBody.data
+
+  // Validate all keys inside.
+  try {
+    await Promise.all(publicKeys.map((key) => verifyRequestBody(key)))
+  } catch (err) {
+    if (err instanceof KnownError) {
+      return respond(err.statusCode, err.responseJson)
+    }
+
+    return respond(400, {
+      error:
+        'Failed to validate public keys: ' +
+        (err instanceof Error ? err.message : `${err}`),
+    })
+  }
+
+  // Validate all keys inside are different from each other.
+  const keySet = new Set(publicKeys.map((key) => key.data.auth.publicKey))
+  if (keySet.size !== publicKeys.length) {
+    return respond(400, {
+      error: 'Keys must be unique.',
+    })
+  }
```

Comment:
Oy, can we wrap this in a method with a descriptive name, like `areAllItemsUnique(...items) => bool`? Like, I'm sure this is right, but it took me more than a second, and having (or later introducing) some boneheaded error in this route would be tragic.